modules/ha-cluster/variables.tf

```
@@ -0,0 +1,328 @@
+  default     = null
+}
+
+variable "docker_password" {
+  type        = string
+  description = "Password (or token) for Docker Hub authentication."
+  default     = null
+}
+
+variable "lb_enabled" {
+  description = "Whether to create an ALB for the server task. Useful for accessing the UI."
+  type        = bool
+  default     = true
+}
+
+variable "internal_alb_listener" {
+  type        = bool
+  description = "ALB is internal-only to reduce load-test costs. When false the ALB will be accessible over the public network."
+  default     = true
+}
+
+variable "deploy_efs_cluster" {
+  type        = bool
+  description = "Deploy EFS Cluster for Consul data storage? Default is true."
+  default     = true
+}
+
+variable "ecs_cluster_name" {
+  description = "Specify an ECS cluster name to deploy the consul services."
+  default     = null
+}
+
+variable "create_ecs_log_group" {
+  description = "Create an ECS Log Group for the containers. Default is True"
+  default     = true
+}
+
+variable "ecs_log_retention_period" {
+  description = "Specify a log retnetion period in days. Default is 7."
+  default     = 7
+}
+
+variable "consul_container_cpu" {
+  description = "Set the Consul Container total CPU limit. Default is 2048."
+  default     = 2048
+}
+
+variable "consul_container_memory" {
+  description = "Set the Consul Cotainer total Memory limit. Default is 4096."
+  default     = 4096
+}
+
+variable "consul_task_cpu" {
+  description = "Set the Consul Server task CPU limit. Default is 1792."
+  default     = 1792
+}
+
+variable "consul_task_memory" {
+  description = "Set the Consul Server container Memory limit. Default is 3584."
+  default     = 3584
+}
+
+variable "datadog_task_cpu" {
+  description = "Set the Consul Server task CPU limit. Default is 1792."
+  default     = 256
+}
+
+variable "datadog_task_memory" {
+  description = "Set the Consul Server container Memory limit. Default is 3584."
+  default     = 512
+}
+
+variable "aws_auto_join" {
+  description = "Enable AWS Auto-Join based on ECS Tag value."
+  default     = true
+}
+
+variable "raft_multiplier" {
+  description = "The Consul Performance Raft-Multiplier. Default is 1."
+  default     = 1
+}
+
+variable "run_k6" {
+  type    = bool
+  default = false
+}
+
+variable "k6_apikey" {
+  description = "K6 API key"
+  type        = string
+  default     = ""
+}
+
+variable "datadog_apikey" {
+  description = "Datadog API key"
+  type        = string
+  default     = ""
+}
+
+variable "lb_ingress_rule_cidr_blocks" {
+  description = "CIDR blocks that are allowed access to the load balancer."
+  type        = list(string)
+  default     = null
+}
+
+variable "lb_ingress_rule_security_groups" {
+  description = "Security groups that are allowed access to the load balancer."
+  type        = list(string)
+  default     = null
+}
+
+variable "consul_count" {
+  description = "Number of consul servers to deploy. Default 3"
+  default     = 3
```

minor / non-blocking: Validate the consul_count is odd and greater than 0. Something like the following:

```suggestion
  default     = 3
  
  validation {
    error_message = "The consul_count must be odd and must be greater than 0."
    condition     = var.consul_count % 2 == 1 && var.consul_count > 0
  }
```